build-theme-previews.py

```diff
@@ -175,6 +172,55 @@ def build_theme_previews(theme_root, samples_root, output_root, screenshot_root)
     return success, fail
 
 
+def i18n_workaround(theme_path):
+    """Installing the i18n extension is tricky and outside of scope."""
+    for subdir, dirs, files in os.walk(theme_path):
+        for template_file in files:
+            if template_file.endswith(".html"):
+                full_path = os.path.join(subdir, template_file)
+                monoglot_template = ""
+                lines = read_mystery_encoding(full_path)
+                for line in lines:
+                    n_line = re.sub(r'_\(([^)]+)\)', r'\1', line)
```

Comment:
Umm, I'm going to say: no. Let's leave aside the fact that this is error prone. You're modifying the theme itself and at this point this is no longer a preview of *that theme*, but maybe for a fork at best.

Reply:
You are replacing `_(...)` with `...`, and without any regard to where it is or what it does I might add. It's an attempt to fix the symptom, not the cause. It may fix some things *now*, no guarantee it will work in the future or what other side effects it might cause.

Reply:
I'd be far more OK if it actually attempted to solve the cause. They need a valid Jinja i18n environment (see [this comment](https://github.com/getpelican/pelican-themes/issues/460#issuecomment-269821765) for a simple implementation). So an easier solution is to create new config file under samples folder, that builds on the default one but with added plugin functionality. Something roughly like this:

```python
extra_code = '''
class i18n(object):
    ...

JINJA_ENVIRONMENT = {'extensions': ['jinja2.ext.i18n']}
PLUGINS = [i18n()]'''

with open('pelican.conf.py') as infile:
    with open('pelican.conf.modified.py') as outfile:
        outfile.write(infile.read() + extra_code)
```

Reply:
By now I've forgotten how I attempted to "solve the cause" of `i18n` environment. I followed countless examples given online and while that isn't a criticism of your suggestion, if you wanted to do that, I think you'd have done it already or at least have more confidence in your suggestion.

In regard to your suggestion that I blindly removed the substring `_(...)` because *it made tests pass* I can assure you that is not the case. Do *you* know what it does? I shall explain.

It contains identifiers for internationalisation. That might matter to you, or not. Personally I don't have the bandwidth to translate my content by hand and various online tools automate the process without a website maintainer needing to add to their burden.

As such I imagined the library to be deprecated. I can't prove a negative, that it doesn't work, just that it isn't something I would use. I can't guarantee my fix is 100% future proof but then, as your test shows 50/138 themes weren't future proof, and I wouldn't be adding to that list. Quite the opposite.

Reply:
Did I offend you? If so, I'm sorry.

I didn't say you don't know what you're doing. However, *that code* is blindly replacing `_(...)` with `...`. I'd like think I know how to read python code by now.

The reason I didn't delve deeper to fix these issues is simply lack of time. All I wanted to do is, since you are motivated to solve this (thanks again, btw), provide assistance to come to a suitable solution. If you don't like my suggestion or don't want to implement it, that's fine. I suppose I can do it myself, if I get a chance.